src/expressions.md

```diff
@@ -600,6 +600,12 @@ within this loop may exit out of this loop or return control to its head.
 See [break expressions](#break-expressions) and [continue
 expressions](#continue-expressions).
 
+The return value of a `loop` expression is `!`, which can be coerced to any
```

Comment:
Instead of saying that `!` can be coerced to any type here, perhaps just link to the section on diverging functions.